enclave_wrangler/objects_api.py

```diff
@@ -1024,6 +1045,118 @@ def get_projects(verbose=False) -> List[Dict]:
+    todo: refactor to do all csets in 1 audit query each, and 1 update_objects_in_db() call
+    todo: performance: Ideally would update all csets with a single query, but right now stability is more important. We
+     want to update one at a time, and as soon as a single case fails, we want to throw an err message at that time.
+     This applies to the 'for cset in cset_objs' part. In order to run update_objects_in_db() on all cset_objs at once,
+     I should also update the audit_query to run on multiple objs at once.
+    """
+    conn = con if con else get_db_connection()
+    cset_objs = cset_objs if isinstance(cset_objs, list) else [cset_objs]
+
+    if set_missing_fields_to_null:
+        cset_objs = cset_objs_set_missing_fields_to_null(cset_objs, schema)
+
+    audit_query = """INSERT INTO code_sets_audit
+        SELECT *, CURRENT_TIMESTAMP AS update_timestamp
+        FROM code_sets
+        WHERE codeset_id = cset_id;"""
+    for cset in cset_objs:
+        cset_id = str(cset['codesetId'])
+        run_sql(con, audit_query.replace('cset_id', cset_id))  # backup to audit table: original
+        update_objects_in_db(con, 'OMOPConceptSet', [cset])
+        run_sql(con, audit_query.replace('cset_id', cset_id))  # backup to audit table: updated
+    if not con:
+        conn.close()
+
+
+# TODO: set check_even_if_still_draft to False at end
+#  - and add a docstring for it
+def clean_improper_finalized_draft_metadata(check_even_if_still_draft=True):
+    """Identify finalized drafts with improper metadata and update their metadata, title and all.
+
+    Checks if title contains '[DRAFT]' and is_draft=false. Updates all metadata.
+    :param check_even_if_still_draft: There have been cases in the past where we did not properly set is_draft to False
+     when the csets was finalized. Setting this to True will check these cases.
+    todo: will this be obsolete when sync_updated_metadata() is complete?
+    """
+    draft_cond = '' if check_even_if_still_draft else " AND is_draft = false"
+    qry = f"""SELECT * FROM code_sets WHERE concept_set_version_title LIKE '%[DRAFT]%'{draft_cond};"""
+    with get_db_connection() as con:
+        csets_db: List[Dict] = [dict(x) for x in sql_query(con, qry)]
+        csets_enclave: List[Dict] = []
+        for cset in csets_db:
+            csets_enclave.append(fetch_cset_version(cset['codeset_id']))
+            # todo: I don't think i need this
+            # cset['properties']['isDraft'] = True
+            # cset['properties']['title'] = cset['properties']['title'].replace('[DRAFT]', '').strip()
+        # TODO: Filter only by those which changed
+
+        csets_enclave2 = cset_objs_set_missing_fields_to_null(csets_enclave)
+        csets_enclave_table_objs: List[Dict] = [convert_row('OMOPConceptSet', 'code_sets', x) for x in csets_enclave2]
+        cset_table_keys_not_in_objs: Set[str] = set(
```

Comment:
## Missing fields from enclave objects?
@Sigfried FYI, these field are in `code_sets` but do not come back when we query the enclave. Do you think it's because these 700 csets I fetched just didn't have any data on these fields? Or are these not available via the enclave API? If not, then how did these get in our `code_sets` table? From datasets API?

- `atlas_json`
- `authoritative_source`
- `comments`
- `date_submitted_zenodo`
- `first_author`
- `has_review`
- `issues`
- `reviewed_by`
- `status`